lib/methods/query.js

```diff
@@ -53,10 +53,10 @@ exports.query = function (Model, options, callback) {
 		orderQuery = orderQuery.slice(0, -1);
 	}
 
-	pagingQuery += ' ' + (whereQuery ? 'AND' : 'WHERE') + ' ROWNUM >= ' + (+options.skip || 0) + ' AND ROWNUM <= ' + (+options.limit);
+	pagingQuery += ' OFFSET ' + (+options.skip || 0) + ' ROWS FETCH NEXT ' + (+options.limit || 10) + ' ROWS ONLY';
 
-	var query = 'SELECT ' + keys + ' FROM ' + table + whereQuery + pagingQuery + orderQuery;
-	this._query(query, values, callback, function (rows, results) {
+	var query = 'SELECT ' + keys + ' FROM ' + table + whereQuery + orderQuery + pagingQuery;
+	this._query(query, values, callback, function (rows) {
```

Comment:
Shouldn't we pass useResultSet here? The limit option in the pagingQuery allows us to specify more than the default of 1000 for maxRows which would result in unexpected behaviour.